bin/si-gen-cloud-control/src/commands/generateSiSpecs.ts

```diff
@@ -0,0 +1,55 @@
+import { getServiceByName, loadCfDatabase } from "../cfDb.ts";
+import { pkgSpecFromCf } from "../specPipeline.ts";
+import { PkgSpec } from "../bindings/PkgSpec.ts";
+import { generateAssetFuncs } from "../pipeline-steps/generateAssetFuncs.ts";
+import {
+  generateSocketsFromDomainProps,
+} from "../pipeline-steps/generateSocketsFromDomainProps.ts";
+
+export function generateSiSpecForService(serviceName: string) {
+  const cf = getServiceByName(serviceName);
+  return pkgSpecFromCf(cf);
+}
+
+export async function generateSiSpecs() {
+  const db = await loadCfDatabase();
+
+  let imported = 0;
+  const cfSchemas = Object.values(db);
+
+  let specs = [] as PkgSpec[];
+
+  for (const cfSchema of cfSchemas) {
+    try {
+      const pkg = pkgSpecFromCf(cfSchema);
+
+      specs.push(pkg);
+    } catch (e) {
+      console.log(`Error Building: ${cfSchema.typeName}: ${e}`);
+    }
+  }
+
+  // EXECUTE PIPELINE STEPS
+  specs = generateSocketsFromDomainProps(specs);
+  specs = generateAssetFuncs(specs);
```

Comment:
This is a great change. My only nit is that I think it's weird we do the bulk of the work (generating the domain, resource, and related inputs) in the loop and then call the rest pipeline steps. I feel like we should break creating the domain and the resource into their own pipeline steps. I also think step should work over a single spec and we should iterate over each one here instead of internally to the step itself. We can talk about it.

Reply:
Some steps will need to take a look at all specs for every spec, like generating the subschemas, so to give each step maximum flexibility, looping internally is necessary

Reply:
But yeah we should break the initial step as much as we can, this is just a first pass. Although It maybe is enough for first release? We'll see